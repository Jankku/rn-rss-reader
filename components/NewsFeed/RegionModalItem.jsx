import { memo } from 'react';
import { useTheme } from '@react-navigation/native';
import { Text, Pressable } from 'react-native';
import { stringOrPlaceholder } from '../../utils/stringutils';

function RegionModalItem({ name, onPress }) {
  const { colors } = useTheme();

  return (
    <Pressable android_ripple={{ color: colors.border }} onPress={onPress}>
      <Text
        style={{
          color: colors.text,
          paddingHorizontal: 16,
          paddingVertical: 12,
        }}
      >
        {stringOrPlaceholder(name)}
      </Text>
    </Pressable>
  );
}

export default memo(RegionModalItem);
