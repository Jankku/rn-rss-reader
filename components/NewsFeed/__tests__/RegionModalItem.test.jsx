import RegionModalItem from '../RegionModalItem';
import { render, fireEvent } from '@testing-library/react-native';

describe('Test RegionModalItem', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<RegionModalItem name={'North Ostrobothnia'} onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should call onPress', () => {
    const onPress = jest.fn();
    const { getByText } = render(<RegionModalItem name={'North Ostrobothnia'} onPress={onPress} />);
    const item = getByText('North Ostrobothnia');

    fireEvent.press(item);
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
