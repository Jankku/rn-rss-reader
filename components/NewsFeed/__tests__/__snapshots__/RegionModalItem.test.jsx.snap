// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RegionModalItem Should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
  >
    North Ostrobothnia
  </Text>
</View>
`;
